src/components/ProductListPage/ProductListPage.js

```diff
@@ -0,0 +1,85 @@
+import { useEffect, useState, useRef } from "react";
+import productCategories from "../../mocks/en-us/product-categories.json";
+import productsMock from "../../mocks/en-us/products.json";
+import Card from "../Card/Card";
+
+const ProductListPage = () => {
+  const [isLoading, setLoading] = useState(true);
+  const initProducts = productsMock.results;
+  const [products, setProducts] = useState(initProducts);
+  const [filters, setFilters] = useState(new Map());
+
+  const handleChange = (event) => {
+    const isChecked = event.target.checked;
+    const value = event.target.value.toLowerCase();
+
+    setFilters((map) => {
+      if (!isChecked) map.delete(value);
+      else map.set(event.target.value.toLowerCase(), event.target.checked);
+
+      return new Map(map);
+    });
+  };
```

Comment:
Please use descriptive names, so other people can understand your code.

```suggestion
  const handleChange = (event) => {
    const isChecked = event.target.checked;
    const categoryName = event.target.value;

    setFilters((prevFilters) => {
      if (!isChecked) prevFilters.delete(categoryName);
      else prevFilters.set(categoryName, true);

      return new Map(prevFilters);
    });
  };
```

Reply:
done